pkg/vault/test/authorizer/TimelockAuthorizerExecute.test.ts

```diff
@@ -426,4 +495,175 @@ describe('TimelockAuthorizer execute', () => {
       await expect(authorizer.cancel(id, { from: other })).to.be.revertedWith('SENDER_IS_NOT_CANCELER');
     });
   });
+
+  describe('getScheduledExecutions', () => {
+    const delay = DAY * 5;
+
+    sharedBeforeEach('set authorizer permission delay', async () => {
+      // We must set a delay for the `setAuthorizer` function as well to be able to give one to `protectedFunction`,
+      // which it must have in order to be able to schedule calls to it.
+      const setAuthorizerAction = await actionId(vault, 'setAuthorizer');
+      await authorizer.scheduleAndExecuteDelayChange(setAuthorizerAction, 2 * delay, { from: root });
+    });
+
+    sharedBeforeEach('grant permission', async () => {
+      const action = await actionId(authenticatedContract, 'protectedFunction');
+      await authorizer.grantPermission(action, user, authenticatedContract, { from: root });
+
+      await authorizer.scheduleAndExecuteDelayChange(action, delay, { from: root });
+    });
+
+    const schedule = async (functionData: string): Promise<number> => {
+      return authorizer.schedule(
+        authenticatedContract.address,
+        authenticatedContract.interface.encodeFunctionData('protectedFunction', [functionData]),
+        [],
+        { from: user }
+      );
+    };
+
+    // We will schedule 5 calls to `protectedFunction`, with data 0x00, 0x01, 0x02, etc.
+    const SCHEDULED_ENTRIES = 5;
+    // There'll be two extra entries: the one used to set the delay for setAuthorizer, and the one used to set the delay
+    // for `protectedFunction`.
+    const TOTAL_ENTRIES = SCHEDULED_ENTRIES + 2;
+
+    sharedBeforeEach('schedule multiple entries', async () => {
+      for (let i = 0; i < SCHEDULED_ENTRIES; ++i) {
+        await schedule(`0x0${i}`);
+      }
+    });
+
+    it('returns the number of total entries', async () => {
+      expect(await authorizer.instance.getScheduledExecutionsCount()).to.equal(TOTAL_ENTRIES);
+    });
+
+    it('reverts if the skip value is too large', async () => {
+      await expect(authorizer.instance.getScheduledExecutions(TOTAL_ENTRIES, 10, false)).to.be.revertedWith(
+        'SKIP_VALUE_TOO_LARGE'
+      );
+    });
+
+    it('reverts if the maxSize value is zero', async () => {
+      await expect(authorizer.instance.getScheduledExecutions(0, 0, false)).to.be.revertedWith('ZERO_MAX_SIZE_VALUE');
+    });
+
+    context('in chronological order', () => {
+      const reverseOrder = false;
+
+      it('returns entries in chronological order', async () => {
+        const entries = await authorizer.instance.getScheduledExecutions(0, 10, reverseOrder);
```

Comment:
Here and below: maybe make 0 and 10 (or at least 10) a named constant, and expect `MAX_SIZE` > `TOTAL_ENTRIES` to document which use case we are in.